File: BankSystem/Domains/Transactions.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BankSystem.Domains
{
    public class Transaction
    {
        public int Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? SenderAccount { get; set; }
        public string? ReceiverAccount { get; set; }
        public decimal Amount { get; set; }
        public string Reason { get; set; }
        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
        public Transaction(string FirstName, string LastName, string SenderAccount ,string ReceiverAccount, decimal Amount, string Reason, DateTimeOffset CreatedAt)
        {
            this.FirstName = FirstName;
            this.LastName = LastName;
            this.SenderAccount = SenderAccount;
            this.ReceiverAccount = ReceiverAccount;
            this.Amount = Amount;
            this.Reason = Reason;
            this.CreatedAt = CreatedAt;
            
        }
    }
}

File: fontendBank/src/app/components/transactions/transactions.ts
import { CommonModule } from '@angular/common';
import { ChangeDetectorRef, Component } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpClient, HttpClientModule, provideHttpClient } from '@angular/common/http';
import { error, log } from 'console';
import { Observable } from 'rxjs/internal/Observable';
interface Transactionss{

 firstName:string
 lastName :string
 senderAccount :string
 receiverAccount :string
 amount :number
 reason :string
 createdAt: Date
}
@Component({
  selector: 'app-transactions',
  imports: [CommonModule, FormsModule, HttpClientModule],
  standalone: true,
  templateUrl: './transactions.html',
  styleUrl: './transactions.css',
})

export class Transactions {
  constructor(private http: HttpClient, private cd: ChangeDetectorRef){}
   transactions$!: Observable<Transactionss[]>;
  
   ngOnInit(): void {
        this.transactions$ =
      this.http.get<Transactionss[]>('https://localhost:40443/api/transactions');
  }
}
